src/fluree/db/ledger/transact/core.clj

```diff
@@ -355,16 +358,20 @@
                     (recur acc** r))
 
                   (or tempid? (tempid/TempId? obj*))
-                  (-> acc
-                      (update :_temp-flakes conj (flake/->Flake _id** pid obj* t true nil))
-                      (recur r))
+                  (do
+                    (swap! upserts conj _id**) ; make sure we retract the old value
```

Comment:
Looks great - so the only condition this issue happened for was upserts?

Meaning, updating a ref on an existing subject that uses an _id value for an existing identity (e.g. `123435456` or two-tuple `[_user/username 'bplatz']`) was working fine?

Reply:
No, and this is part of why I'm not sure this is the best / right solution.

Updating refs w/ two-tuples was *not* working; at least in the case where it was a tempid for something being created in the same txn; which is what my test was doing. That's an "upsert" in the sense that because it's not marked `:multi true`, it should update the existing value, but instead it was being added to the predicate's object as if it were `:multi true`. But none of the preds are marked `:upsert true`, so I wasn't sure if the semantics were quite right here. Is there a semantic difference between `upsert` and `update` for an existing subject?

I'm working with [these test preds](https://github.com/fluree/ledger/blob/59697cc698def9021f9a65b8c54b923dc9991287/test/fluree/db/ledger/docs/transact/transactions.clj#L39-L52).

I can write some additional tests for existing ids, etc.

Reply:
Gaining more insight as I write additional tests:

Referencing a pre-existing subject w/ a two-tuple *was* working before this change (and still does afterward, just to be clear). So far it seems that the only broken case was referencing a subject in the same transaction with a tempid.

But I'm going to write at least one more test for referencing an existing subject by id instead of two-tuple.

Reply:
And referencing pre-existing subjects by id worked and works too.